docs/06/01/how-it-works.md

```diff
@@ -0,0 +1,73 @@
+## How the Component Adapter Works
+
+The Component Adapter uses React's Context API to provide UI components to the SDK. At its core, the system consists of:
+
+1. **ComponentsContext** - A React context that holds references to all UI components ([View on GitHub](https://github.com/Gusto/embedded-react-sdk/blob/main/src/contexts/ComponentAdapter/useComponentContext.ts))
+2. **ComponentsProvider** - A provider component that makes custom UI components available throughout the component tree ([View on GitHub](https://github.com/Gusto/embedded-react-sdk/blob/main/src/contexts/ComponentAdapter/ComponentsProvider.tsx))
+3. **useComponentContext** - A hook to access the components within SDK components ([View on GitHub](https://github.com/Gusto/embedded-react-sdk/blob/main/src/contexts/ComponentAdapter/useComponentContext.ts))
+
```

Comment:
Wondering if we should actually just omit this? worried about trying to keep these implementation details up to date if we ever update our approach